src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc

```diff
@@ -95,8 +95,12 @@ void AddSubMenuRequest::Run() {
       (*message_)[strings::msg_params][strings::menu_name];
   msg_params[strings::app_id] = app->app_id();
 
-  if (((*message_)[strings::msg_params].keyExists(strings::sub_menu_icon)) &&
-      CheckSubMenuIcon()) {
+  if (((*message_)[strings::msg_params].keyExists(strings::sub_menu_icon))) {
+    if (!CheckSubMenuIcon()) {
+      LOG4CXX_ERROR(logger_, "Sub-menu icon is not valid.");
```

Comment:
@LevchenkoS If icon does not exist then it is ok. If icon exists and invalid then we should send INVALID_DATA.